tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -5349,14 +5349,16 @@ void SpirvEmitter::condenseVectorElementExpr(
     const HLSLVectorElementExpr *expr, const Expr **basePtr,
     hlsl::VectorMemberAccessPositions *flattenedAccessor) {
   llvm::SmallVector<hlsl::VectorMemberAccessPositions, 2> accessors;
-  accessors.push_back(expr->getEncodedElementAccess());
+  *basePtr = expr;
 
   // Recursively descending until we find the true base vector. In the
```

Comment:
nit: I understand this code, but if we added a comment to explain the meaning of "true base vector", it would be more clear and easier to understand. "true base vector" means "the base vector that does not have a base vector (or that is not an element of other vector)" right?

Reply:
yes.
If we have something like `myVector.xyz.xx.yz`, then there will be 3 `HLSLVectorElementExpr`s.
(myVector.xyz)
then
(result.xx)
then
(result.yz)

And if you start from the bottom (.yz), then the true base is `myVector`

Reply:
I'll add a comment.